homeassistant/components/flux_led/light.py

```diff
@@ -253,75 +253,69 @@ def effect(self):
         for effect, code in EFFECT_MAP.items():
             if current_mode == code:
                 return effect
-
         return None
 
     async def async_turn_on(self, **kwargs):
         """Turn the specified or all lights on and wait for state."""
         await self.hass.async_add_executor_job(partial(self._turn_on,
                                                        **kwargs))
-        # The bulb needs a second to tell its new values,
-        # so we wait 2 seconds before updating
-        await sleep(2)
+        # The bulb needs a bit to tell its new values,
+        # so we wait 1 second before updating
+        await sleep(1)
 
     def _turn_on(self, **kwargs):
         """Turn the specified or all lights on."""
-        self._bulb.turnOn()
+        if not self.is_on:
+            self._bulb.turnOn()
 
         hs_color = kwargs.get(ATTR_HS_COLOR)
         brightness = kwargs.get(ATTR_BRIGHTNESS)
         effect = kwargs.get(ATTR_EFFECT)
         white = kwargs.get(ATTR_WHITE_VALUE)
 
-        # Show warning if effect set with rgb, brightness, or white level
-        if effect and (brightness or white or hs_color):
-            _LOGGER.warning("RGB, brightness and white level are ignored when"
-                            " an effect is specified for a flux bulb")
-
-        # Random color effect
-        if effect == EFFECT_RANDOM:
-            self._bulb.setRgb(random.randint(0, 255),
-                              random.randint(0, 255),
-                              random.randint(0, 255))
-            return
-
-        if effect == EFFECT_CUSTOM:
-            if self._custom_effect:
-                self._bulb.setCustomPattern(
-                    self._custom_effect[CONF_COLORS],
-                    self._custom_effect[CONF_SPEED_PCT],
-                    self._custom_effect[CONF_TRANSITION])
+        if all(None for item in [hs_color, brightness, effect, white]):
```

Comment:
That looks weird, it will never be true? You probably meant it like this:
```python
        if all(item is None for item in [hs_color, brightness, effect, white]):
```